packages/contracts/contracts/optimistic-ethereum/bridge/MockL1CrossDomainMessenger.sol

```diff
@@ -0,0 +1,43 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+/* Contract Imports */
+import { L1CrossDomainMessenger } from "./L1CrossDomainMessenger.sol";
+
+/**
+ * @title MockL1CrossDomainMessenger
+ */
+contract MockL1CrossDomainMessenger is L1CrossDomainMessenger {
```

Comment:
So is it possible to just have a MockCrossDomainMessenger like before and not have them inherit from the more complete L1 / L2 messengers?

Basically we don't mock whole async 'relay' functionality in the mock version of the tests because that is all handled in the background for users / dapps

Reply:
So these mocks bypass the proof checks within the complete contracts. It keeps the interface the same and does minimal verification so that developers can sanity check for common cases (e.g., attempting to send the same message twice). I feel like it's a good middle ground for now.

Reply:
> so that developers can sanity check for common cases (e.g., attempting to send the same message twice).

Do you think that developers should write unit tests which call `replayMessage(...)` ? My guess would be that this is not really required in the same way that developers normally don't add tests for users who send a tx with under-priced gas & then use a replace tx with higher gas price.

Generally down to keep this interface but I just want to make sure that we're on the same page wrt what makes sense to put in unit tests & what doesn't

Reply:
To me the biggest thing missing in our mocks is a helper function which you can import called `waitForCrossDomainMessage(txHash)` which can be used something like this:

```
// Do something which sends a message to L2
const txHash = await myL1Contract.sendDepositToL2()
// Wait for the L1->L2 message to be processed
await waitForCrossDomainMessage(txHash)
// Do another thing on L2 like spend the deposit
await myL2Contract.spendMyMoney()
```